.github/scripts/junit.py

```diff
@@ -0,0 +1,175 @@
+                cur_suite.test_failures.append(failure)
+                #print(f"{cur_suite}#{cur_test} {elem.attrib}: {elem.text}")
+            elif elem.tag == "skipped":
+                skipped = partial_test_case(None, None, None)
+                cur_suite.skipped_tests.append(skipped)
+            else:
+                pass
+        elif event == "end":
+            stack.pop()
+            if elem.tag == "testsuite":
+                yield cur_suite
+                cur_suite = None
+                partial_test_failure = None
+        else:
+            logger.error(f"Unhandled xml event {event}: {elem}")
+
+
+def pretty_time_duration(seconds: float) -> str:
+    time_min, time_sec = divmod(int(seconds), 60)
+    time_hour, time_min = divmod(time_min, 60)
+    time_fmt = ""
+    if time_hour > 0:
+        time_fmt += f"{time_hour}h"
+    if time_min > 0:
+        time_fmt += f"{time_min}m"
+    time_fmt += f"{time_sec}s"
+    return time_fmt
+
+
+if __name__ == "__main__":
+    """
+    Parse JUnit XML reports and generate GitHub job summary in Markdown format.
+
+    Exits with status code 0 if no tests failed, 1 otherwise.
+    """
+    if not os.getenv("GITHUB_WORKSPACE"):
+        print("This script is intended to by run by GitHub Actions.")
+        exit(1)
+
+    reports = glob(pathname="**/test-results/**/*.xml", recursive=True)
+    logger.debug(f"Found {len(reports)} JUnit results")
+    workspace_path = get_env("GITHUB_WORKSPACE") # e.g., /home/runner/work/apache/kafka
+
+    total_file_count = 0
+    total_tests = 0
+    total_skipped = 0
+    total_failures = 0
+    total_errors = 0
+    total_time = 0
+    table = []
+    for report in reports:
+        with open(report, "r") as fp:
+            logger.debug(f"Parsing {report}")
+            for suite in parse_report(workspace_path, report, fp):
+                total_tests += suite.tests
+                total_skipped += suite.skipped
+                total_failures += suite.failures
+                total_errors += suite.errors
+                total_time += suite.time
+                for test_failure in suite.test_failures:
+                    logger.debug(f"Found test failure: {test_failure}")
+                    simple_class_name = test_failure.class_name.split(".")[-1]
+                    table.append(("❌", simple_class_name, test_failure.test_name, test_failure.failure_message, f"{test_failure.time:0.2f}s"))
+                for skipped_test in suite.skipped_tests:
+                    simple_class_name = skipped_test.class_name.split(".")[-1]
+                    logger.debug(f"Found skipped test: {skipped_test}")
+                    table.append(("⚠️", simple_class_name, skipped_test.test_name, "Skipped", ""))
+    duration = pretty_time_duration(total_time)
+    print(f"{total_tests} tests run in {duration}, {total_failures} failed ❌, {total_skipped} skipped ⚠️, {total_errors} errors.")
```

Comment:
Could we add the artifact url link? contributors can download the total reports when they check this great summary? the link (https://github.com/actions/upload-artifact/issues/50#issuecomment-1856471599) shows how to get the artifact url and we can pass it through env. WDYT?